WalletWasabi/WabiSabi/Backend/Rounds/Utils/CoinJoinTransactionArchiver.cs

```diff
@@ -0,0 +1,64 @@
+using NBitcoin;
+using System;
+using System.IO;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace WalletWasabi.WabiSabi.Backend.Rounds.Utils
+{
+	public class CoinJoinTransactionArchiver
+	{
+		private static readonly JsonSerializerOptions Options = new()
+		{
+			WriteIndented = true, // Pretty print.
+		};
+
+		public CoinJoinTransactionArchiver(string directoryPath)
+		{
+			DirectoryPath = directoryPath;
+			IoHelpers.EnsureDirectoryExists(directoryPath);
+		}
+
+		/// <summary>Path where to store transactions.</summary>
+		public string DirectoryPath { get; }
+
+		/// <summary>
+		/// Save CoinJoin transaction in a JSON file for later inspection.
+		/// </summary>
+		/// <param name="transaction">Transaction to store.</param>
+		/// <param name="currentDate">Date when the transaction was created or <c>null</c> to use current UTC time.</param>
+		/// <returns>Full path to the stored file.</returns>
+		public async Task<string> StoreJsonAsync(Transaction transaction, DateTimeOffset? currentDate = null)
+		{
+			currentDate ??= DateTimeOffset.UtcNow;
+
+			long unixTimeStampMs = currentDate.Value.ToUnixTimeMilliseconds();
+			string txBytes = transaction.ToHex();
+
+			// Serialize entry.
+			TransactionInfo entry = new(unixTimeStampMs, transaction.GetHash().ToString(), txBytes);
+			string json = JsonSerializer.Serialize(entry, Options);
+
+			// Add UNIX timestamp to the file name to let the files be sorted by date by default.
+			string fileName = $"CoinJoinTransaction.{unixTimeStampMs}.{transaction.GetHash()}.json";
```

Comment:
[Current timestamp](https://www.unixtimestamp.com/) is 1632826224 (in seconds). 2632826224 corresponds with Fri Jun 06 2053 12:37:04 GMT+0000.

Reply:
> [Current timestamp](https://www.unixtimestamp.com/) is 1632826224 (in seconds). 2632826224 corresponds with Fri Jun 06 2053 12:37:04 GMT+0000.

good to know (though 1 -> 2 is not a problem, 9 -> 10 is what i was referring to)

Reply:
The time interval for 1 -> 2 and 9 -> 10 is the same, isn't it?

Reply:
sure but we have much longer from 1 -> 10... even though 1 -> 2 is already way more than enough than we should worry about.

none of this matters because you already applied my feedback to use ISO 8601 as a human readable and lexicographically sortable date format, but TIL unix timestamps should remain lexicographically sortable even without leading 0s for a good long time